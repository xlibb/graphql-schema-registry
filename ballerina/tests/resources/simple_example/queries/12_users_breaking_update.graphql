mutation PublishSubgraph {
    publishSubgraph(
        schema: {
            name: "users"
            url: "http://localhost:8081"
            schema: """
extend schema @link(url: "https://specs.apollo.dev/federation/v2.0", import: ["@key"]) 

type Query {
  "Returns the list of users"
  users: [User!]!
}

"Represents a user in the system"
type User @key(fields: "id") {
  "The unique identifier for the user"
  id: ID!
  "The name of the user"
  name: String!
}
            """
        }
        isForced: false
    ) {
        schema
        apiSchema
        version
        hints
        diffs {
            severity
            action
            subject
            location
            value
            fromValue
            toValue
        }
    }
}
