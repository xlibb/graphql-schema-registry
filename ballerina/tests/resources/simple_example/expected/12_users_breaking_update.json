{
    "errors": [
        {
            "message": "Operation causes breaking changes",
            "locations": [
                {
                    "line": 2,
                    "column": 5
                }
            ],
            "path": [
                "publishSubgraph"
            ],
            "extensions": {
                "errors": [
                    {
                        "message": "BREAKING: Removed 'email' from User.",
                        "details": {
                            "diff": {
                                "severity": "BREAKING",
                                "action": "REMOVED",
                                "subject": "FIELD",
                                "location": [
                                    "User"
                                ],
                                "value": "email",
                                "fromValue": null,
                                "toValue": null
                            }
                        }
                    }
                ]
            }
        }
    ],
    "data": {
        "publishSubgraph": null
    }
}